### PR TITLE
JWT token

### PR DESCRIPTION
**Pull request by Eidenn005**
Added JWT token support
Added UserLoginRequest and UserLoginResponse dtos
Added JwtUtil and Auth service

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -87,6 +87,22 @@
...
+		<dependency>
+			<groupId>io.jsonwebtoken</groupId>
+			<artifactId>jjwt-api</artifactId>
+			<version>0.12.5</version>
+		</dependency>
+		<dependency>
+			<groupId>io.jsonwebtoken</groupId>
+			<artifactId>jjwt-jackson</artifactId>
+			<version>0.12.5</version>
+			<scope>runtime</scope>
```
Comment (andrii-hoienko) [inline pom.xml]: Please use property to define dependecy version

File: src/main/java/com/example/onlinebookstore/dto/UserLoginRequestDto.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+import jakarta.validation.constraints.Size;
+import lombok.Data;
+
+@Data
+public class UserLoginRequestDto {
+    @Email
+    @NotBlank
+    private String email;
+    @NotBlank
+    @Size(min = 8, max = 20)
```
Comment (andrii-hoienko) [inline src/main/java/com/example/onlinebookstore/dto/UserLoginRequestDto.java]: I think password could be much bigger that 20 chars

File: src/main/java/com/example/onlinebookstore/security/JwtAuthenticationFilter.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+            );
+            SecurityContextHolder.getContext().setAuthentication(authentication);
+        }
+        filterChain.doFilter(request, response);
+    }
+
+    private String getToken(HttpServletRequest request) {
+        String bearerToken = request.getHeader(AUTH_HEADER);
+        if (StringUtils.hasText(bearerToken) && bearerToken.startsWith(BEARER_PREFIX)) {
+            return bearerToken.substring(START_INDEX);
```
Comment (andrii-hoienko) [inline src/main/java/com/example/onlinebookstore/security/JwtAuthenticationFilter.java]:
```suggestion
            return bearerToken.substring(BEARER_PREFIX.legth);
```

File: src/main/java/com/example/onlinebookstore/security/JwtAuthenticationFilter.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+import org.springframework.stereotype.Component;
+import org.springframework.util.StringUtils;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+@Component
+@RequiredArgsConstructor
+public class JwtAuthenticationFilter extends OncePerRequestFilter {
+    private static final String AUTH_HEADER = "Authorization";
+    private static final String BEARER_PREFIX = "Bearer ";
+    private static final int START_INDEX = 7;
```
Comment (andrii-hoienko) [inline src/main/java/com/example/onlinebookstore/security/JwtAuthenticationFilter.java]:
```suggestion
```

File: src/main/java/com/example/onlinebookstore/security/JwtUtil.java [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+    public boolean isValidToken(String token) {
+        try {
+            Jws<Claims> claimsJws = Jwts.parser()
+                    .verifyWith(secret)
+                    .build()
+                    .parseSignedClaims(token);
+
+            return !claimsJws.getPayload().getExpiration().before(new Date());
+        } catch (JwtException | IllegalArgumentException e) {
+            throw new JwtException("Expired or invalid JWT token");
```
Comment (andrii-hoienko) [inline src/main/java/com/example/onlinebookstore/security/JwtUtil.java]:
```suggestion
           return false;
```

File: src/main/resources/application.properties [hunk 6]
```diff
@@ -5,4 +5,7 @@ spring.datasource.password=${DB_PASSWORD}
 spring.jpa.properties.hibernate.dialect=org.hibernate.dialect.MySQL8Dialect
 spring.jpa.hibernate.ddl-auto=validate
 spring.jpa.show-sql=true
+jwt.expiration=3600000
+jwt.secret=anotheranothersupersupersupersecretstring1234567890566855248679564
```
Comment (andrii-hoienko) [inline src/main/resources/application.properties]: Please use env variables

**Review by andrii-hoienko — CHANGES_REQUESTED** [review]
Nice, left minor comments

File: pom.xml (hunk 1 above)
Comment (Eidenn005) [reply pom.xml]: done

File: src/main/java/com/example/onlinebookstore/dto/UserLoginRequestDto.java (hunk 2 above)
Comment (Eidenn005) [reply src/main/java/com/example/onlinebookstore/dto/UserLoginRequestDto.java]: done

File: src/main/java/com/example/onlinebookstore/security/JwtAuthenticationFilter.java (hunk 4 above)
Comment (Eidenn005) [reply src/main/java/com/example/onlinebookstore/security/JwtAuthenticationFilter.java]: done

File: src/main/java/com/example/onlinebookstore/security/JwtUtil.java (hunk 5 above)
Comment (Eidenn005) [reply src/main/java/com/example/onlinebookstore/security/JwtUtil.java]: done

File: src/main/resources/application.properties (hunk 6 above)
Comment (Eidenn005) [reply src/main/resources/application.properties]: done

File: src/main/resources/application.properties [hunk 7]
```diff
@@ -5,4 +5,7 @@ spring.datasource.password=${DB_PASSWORD}
 spring.jpa.properties.hibernate.dialect=org.hibernate.dialect.MySQL8Dialect
 spring.jpa.hibernate.ddl-auto=validate
 spring.jpa.show-sql=true
+jwt.expiration=${jwt_expiration}
+jwt.secret=${jwt_secret}
+
 server.port=8081
```
Comment (karma-o) [inline src/main/resources/application.properties:11]: can we change the port back to the default 8080?

**Review by karma-o — APPROVED** [review]
Well done!

File: src/main/resources/application.properties (hunk 7 above)
Comment (Eidenn005) [reply src/main/resources/application.properties:11]: Will try. It was old problem resolving, didnt touch it atm.

**Pull request merged**